dataflux_pytorch/tests/test_dataflux_mapstyle_dataset.py

```diff
@@ -435,6 +429,69 @@ def test_init_with_perm(self):
             f"got max_composite_object_size for compose download{ds.config.max_composite_object_size}, want {want_size}",
         )
 
+    @mock.patch("dataflux_pytorch.dataflux_mapstyle_dataset.dataflux_core")
+    def test_getstate(self, mock_dataflux_core):
+        """Tests that the dataset.__getitems__ method returns the list of the correct downloaded object content."""
+        # Arrange.
+        mock_listing_controller = mock.Mock()
+        mock_listing_controller.run.return_value = self.want_objects
+        mock_dataflux_core.fast_list.ListingController.return_value = (
+            mock_listing_controller)
+
+        # Act.
+        ds = dataflux_mapstyle_dataset.DataFluxMapStyleDataset(
+            project_name=self.project_name,
+            bucket_name=self.bucket_name,
+            config=self.config,
+            storage_client=self.storage_client,
+        )
+
+        want_state = ds.__dict__.copy()
+        want_state.pop("storage_client")
+
+        states = ds.__getstate__()
+
+        # Assert.
+        self.assertEqual(
+            states,
+            want_state,
+            f"got dataflux_mapstyle_dataset params {states}, want {want_state}",
+        )
+
+    @mock.patch("dataflux_pytorch.dataflux_mapstyle_dataset.dataflux_core")
+    def test_setstate(self, mock_dataflux_core):
+        """Tests that the dataset.__getitems__ method returns the list of the correct downloaded object content."""
+        # Arrange.
+        mock_listing_controller = mock.Mock()
```

Comment:
if this little chunk of code is the same everywhere you could move it to a test helper function to reduce repeat code.